src/roiextractors/memmapextractors.py

```diff
@@ -2,109 +2,118 @@
 import shutil
 import tempfile
 from pathlib import Path
+from typing import Union, Optional
 
 import numpy as np
 from tqdm import tqdm
 
 from .extraction_tools import PathType, check_get_frames_args
 from .imagingextractor import ImagingExtractor
 
+PathType = Union[str, Path]
+DtypeType = Union[str, np.dtype]
+OptionalDtypeType = Optional[DtypeType]
+DtypeType = Union[str, np.dtype]
+
 
 class MemmapImagingExtractor(ImagingExtractor):
+
+    extractor_name = "MemmapImagingExtractor"
+
     def __init__(
         self,
-        imaging_extractor: ImagingExtractor,
-        save_path: PathType = None,
-        verbose: bool = False,
+        file_path: PathType,
+        frame_shape: tuple,
+        dtype: DtypeType,
+        offset: int = 0,
+        sampling_frequency: float = 0,
+        image_structure_to_axis: dict = None,
     ):
-        ImagingExtractor.__init__(self)
-        self.imaging = imaging_extractor
-        tmp_folder = self.get_tmp_folder()
-
-        if save_path is None:
-            self._is_tmp = True
-            self._tmp_file = tempfile.NamedTemporaryFile(suffix=".dat", dir=tmp_folder).name
-        else:
-            save_path = Path(save_path)
-            if save_path.suffix != ".dat" and save_path.suffix != ".bin":
-                save_path = save_path.with_suffix(".dat")
-            if not save_path.parent.is_dir():
-                os.makedirs(save_path.parent)
-            self._is_tmp = False
-            self._tmp_file = save_path
-
-        self._save_memmap_video(verbose)
-        self._verbose = verbose
```

Comment:
This doesn't have to be outright removed - it can serve as the @staticmethod `write_imaging` for the class, which is what it seems to do as-is.

Reply:
I don't think it works as it is.

Reply:
Wow, https://github.com/staticmethod is an actual GitHub user, lol

Reply:
> I don't think it works as it is.

The `_save_memmap_video` doesn't work when `save_path` isn't `None`?

Reply:
x D

Reply:
I re-wrote the `write_imaging` method now.